vision_datasets/factory/operations/spawn_factory.py

```diff
@@ -0,0 +1,24 @@
+from ...common import DatasetTypes
+from ...data_manifest import Spawn, SpawnConfig
+from .supported_operations_by_data_type import SupportedOperationsByDataType
+
+
+class SpawnFactory:
+    _mapping = {}
+
+    @classmethod
+    def direct_register(cls, klass, data_type: DatasetTypes):
+        SupportedOperationsByDataType.add(data_type, klass)
+        cls._mapping[data_type] = klass
+        return klass
+
+    @classmethod
+    def register(cls, data_type: DatasetTypes):
+        def decorator(klass):
+            return SpawnFactory.direct_register(klass, data_type)
+        return decorator
+
+    @classmethod
+    def create(cls, data_type: DatasetTypes, config: SpawnConfig, *args, **kwargs) -> Spawn:
```

Comment:
it sounds bit redundant to call factory then call run() method
```
operation = Factory.create(data_type, ...)
operation.run(dataset)
```
Instead, how about we define a function for each operation? then, inside the function, we can find corresponding class and use it.
```
from vision_datasets.operations import spawn
dataset = spawn(dataset, config)
```

Reply:
The Operation base class is supposed to be replaced by Task from irisml in future for easy integration and extendability in future, which is why it is a little bit layered